Dockerfile

```diff
@@ -1,15 +1,15 @@
-FROM golang:1.7.4
+FROM golang:1.9
```

Comment:
Should be `golang:1.9.0`

Reply:
`1.9` is the equivalent of `1.9.x` or `^1.9`

Reply:
yeah, and that's the issue - we should be explicit about versions

if we are not explicit, then we won't know what base image was used to build the result image

Reply:
let's say important security fix lands into 1.9.1..

with `1.9`, there's no guarantee that the builder won't pick up the older image `1.9.0`

Reply:
that's where `docker build --pull` comes in https://docs.docker.com/engine/reference/commandline/build/

Reply:
Yeah, `docker build --pull` is same as running `docker pull && docker build`.

But it still doesn't guarantee that the build will really pick up `1.9.1`, though. It may pick up `1.9.0` or `1.9.2+` based on the state of Docker registry it is connected to. And I assume we might switch off Docker Hub pretty soon.

The problem is that you won't know what version was used to build the production image (async CI job). You won't have the control. You could just guess if the newer tag was already available at the build time - or you could try parsing the image layers to get the SHA of specific version.

Anyway, this is not big deal right now. But we'll need specific versions for Kubernetes in the future for sure.